<?xml version="1.0" ?>
<launch>
    <arg name="config" default="example_scene_ur_intent"/>
    <arg name="root_link" default="base_link"/>
    <arg name="tip_link" default="ee_link"/>
    <arg name="mp_display" default="false"/>
    <arg name="cmp_display" default="false"/>
    <arg name="show_debug_statements" default="false"/>
    <arg name="constraints_disabled" default="false"/>
    <arg name="samples" default="7"/>
    <arg name="deviation" default="0.05"/>

    <!-- Launch UR3 URDF model -->
    <!--include file="$(find ur_e_description)/launch/ur3e_upload.launch"/-->
    <param name="robot_description" command="$(find xacro)/xacro 
        '$(find constrained_manipulability)/urdf/ur3e_with_screwdriver_tool.xacro'"/>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find constrained_manipulability)/rviz/view_robot.rviz" required="true"/>

    <!-- Run intention estimation -->
    <node name="intended_joint_estimator" pkg="intended_joint_estimation" type="intended_joint_estimator.py" output="screen" required="true"/>

    <!-- Launch polytopes -->
    <node name="ik_example" pkg="constrained_manipulability" type="ik_example" output="screen" clear_params="true">
        <rosparam file="$(find constrained_manipulability)/scene_config/$(arg config).yaml" command="load"/>

        <remap from="/joint_traj" to="/scaled_pos_traj_controller/command"/>
        <remap from="/joint_command" to="/joint_group_vel_controller/command"/>

        <param name="root" type="string" value="$(arg root_link)"/>
        <param name="tip" type="string" value="$(arg tip_link)"/>
        <param name="pos_controller" type="string" value="scaled_pos_traj_controller"/>
        <param name="vel_controller" type="string" value="joint_group_vel_controller"/>
        <param name="show_cmp" type="bool" value="$(arg cmp_display)"/>
        <param name="show_mp" type="bool" value="$(arg mp_display)"/>
        <param name="ignore_constraints" type="bool" value="$(arg constraints_disabled)"/>
        <param name="debug_statements" type="bool" value="$(arg show_debug_statements)"/>

        <param name="num_samples" type="int" value="$(arg samples)"/>
        <param name="search_deviation" type="double" value="$(arg deviation)"/>
    </node>
</launch>